### PR TITLE
[JDBC] Simplified Parser

### PR DESCRIPTION
**Pull request by chernser**
## Summary

Closes 
## Checklist
Delete items not relevant to your PR:
- [x] Closes #<issue ref>
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

---

> [!NOTE]
> **Medium Risk**
> Introduces a new selectable SQL parsing path that influences statement classification (`hasResultSet`/`insert`/table extraction), which can affect execution behavior for edge-case queries; changes are covered by extensive new parser tests.
> 
> **Overview**
> Adds a new *lightweight* ANTLR4 grammar (`ClickHouseLexer.g4` + `ClickHouseLightParser.g4`) and a corresponding `SqlParserFacade` option (`SQLParser.ANTLR4_LIGHT`) that **parses only minimal structure** (verb/`USE` db, and `INSERT` table + optional column list) while otherwise accepting arbitrary tokens.
> 
> `ParsedStatement` and `ParsedPreparedStatement` now record `statementVerb`, and the light parser uses this plus a verb allowlist to infer `hasResultSet` without deep parsing; placeholder scanning remains the existing string-based `parseParameters`.
> 
> Tests are updated/added to cover the new parser mode (including quoted identifiers with dots and verb/result-set classification), and the row-binary batch insert integration test is moved from `PreparedStatementTest` to `WriterStatementImplTest` and parameterized to run under multiple parser selections.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 82071299e4ad437a6bda7dff047c0d465041a9b2. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: jdbc-v2/src/main/antlr4/com/clickhouse/jdbc/internal/parser/antlr4_light/ClickHouseLightParser.g4 [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    : USE identifier
+    ;
+
+settingExprList
+    : settingExpr (COMMA settingExpr)*
+    ;
+
+settingExpr
+    : identifier EQ_SINGLE (literal | JDBC_PARAM_PLACEHOLDER)
+    ;
```
[the remark this answers is not in the document]
Comment (chernser) [reply jdbc-v2/src/main/antlr4/com/clickhouse/jdbc/internal/parser/antlr4_light/ClickHouseLightParser.g4:76]: Good self catch =)

File: jdbc-v2/src/test/java/com/clickhouse/jdbc/WriterStatementImplTest.java [hunk 2]
```diff
@@ -28,4 +41,81 @@ public void testTargetTypeMethodThrowException() throws SQLException {
...
+                    assertTrue(rs.getFloat(3) != 0.0f);
+                    assertTrue(rs.getInt(4) != 0);
+                    count++;
+                }
+                assertEquals(count, nBatches);
+
+                stmt.execute("TRUNCATE " + table);
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (chernser) [reply jdbc-v2/src/test/java/com/clickhouse/jdbc/WriterStatementImplTest.java:120]: Will look into it.
